arch/sim/src/sim/up_initialstate.c

```diff
@@ -84,10 +97,10 @@ void up_initial_state(struct tcb_s *tcb)
    * the return address.
    */
 
-  tcb->xcp.regs[JB_SP] = (xcpt_reg_t)tcb->stack_base_ptr +
-                                     tcb->adj_stack_size -
-                                     sizeof(xcpt_reg_t);
+  tcb->xcp.regs[JB_SP] = (xcpt_reg_t)
+      STACK_ALIGN_DOWN((long)tcb->stack_base_ptr + tcb->adj_stack_size - 1);
```

Comment:
let's fix the alignment issue in another patch?

Reply:
Right ... as the basic M1 support in done (PR ready, but not integrated), I'll think about the alignment issue in another PR.

Reply:
let's fix alignment issue first before merge this patch?

Reply:
Yes ...

Reply:
I reviewed code again and the subtraction cannot be removed. There is a specific reason for it (not related to any aligment issue) on the amd64 architecture:

    /* Note: The amd64 ABI requires 16-bytes alignment _before_ a function
     * call.
     * On the other hand, our way to set up and switch to a new context
     * is basically a JUMP.
     * Thus, we need to emulate the effect of a CALL here, by subtracting
     * sizeof(xcpt_reg_t), which is the amount a CALL would move RSP to store
     * the return address.
     */

The subtraction simulates how a CALL would change the stack.

On the ARM64 architecture the return address is not stored on the stack, but in LR register. There's no need for the subtraction on the ARM64 architecture as the stack is unchanged.

I changed the code in a way, that the subtraction is only effective on none ARM64 architectures and added a note.

    * Note: On ARM64 architectures the return address is passed via LR
    *       register. No extra adjustment for the stack needed.

Reply:
> ```
> /* Note: The amd64 ABI requires 16-bytes alignment _before_ a function
>  * call.
>  * On the other hand, our way to set up and switch to a new context
>  * is basically a JUMP.
>  * Thus, we need to emulate the effect of a CALL here, by subtracting
>  * sizeof(xcpt_reg_t), which is the amount a CALL would move RSP to store
>  * the return address.

But the thread never return to the bottom of the stack, I don't believe that it's so critical whether we subtract or not 8bytes.
@ptka and @yamt could you try this PR:
https://github.com/apache/incubator-nuttx/pull/5507.

Reply:
I tested https://github.com/apache/incubator-nuttx/pull/5507 in combination with this test on MacBook Air M1 2020 running Monterey 12.2.1. Executed test: sim:nsh with additional ostest configured. Result: complete ostest passed w/o error.

Reply:
Let's continue the discussion in #5507